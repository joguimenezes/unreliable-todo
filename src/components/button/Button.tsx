import React from 'react'
import styled from 'styled-components';

import COLORS from '../../utils/constants/color.constant';
import MEDIA_QUERIES from '../../utils/constants/mediaQuery.constant';
import Loader from '../loader/Loader';

interface ButtonProps {
  className?: string,
  isLoading: boolean,
  onClick?: (event: React.MouseEvent) => void,
  testId?: string,
  text: string,
}

const Button = ({ 
  className,
  isLoading,
  onClick,
  testId,
  text
}: ButtonProps) => {
  return (
    <Btn
      className={className}
      data-testid={testId}
      onClick={onClick}
    >
      {isLoading
        ? <Loader backgroundColor={COLORS.PRIMARY} />
        : text}
    </Btn>
  )
}

const Btn = styled.button`
  background-color: ${COLORS.SECONDARY};
  border-radius: 99px;
  border: none;
  color: ${COLORS.PRIMARY};
  cursor: pointer;
  height: 55px;
  transition: all .5s;
  width: calc(100% - 25px);

  &:hover {
    opacity: 0.8;
  }

  @media ${MEDIA_QUERIES.DESKTOP_SCREEN} {
    width: 385px;
  }
`;

export default Button