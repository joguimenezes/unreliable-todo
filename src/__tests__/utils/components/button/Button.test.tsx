import React from 'react';
import { render } from '@testing-library/react';

import Button from '../../../../components/button/Button';

describe('Button tests', () => {
  describe('should render correctly', () => {
    it('when is loading', () => {
      const props = {
        isLoading: true,
        text: 'Button',
      };

      const { container } = render(<Button {...props}/>);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('when loaded', () => {
      const props = {
        isLoading: false,
        onClick: jest.fn(),
        text: 'Button'
      };

      const { container } = render(<Button {...props}/>);
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});