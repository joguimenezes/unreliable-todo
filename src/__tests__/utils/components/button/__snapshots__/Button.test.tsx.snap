// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button tests should render correctly when is loading 1`] = `
<button
  class="sc-bdVaJa gXMDQW"
  type="button"
>
  <svg
    height="40px"
    id="loader-1"
    version="1.1"
    viewBox="0 0 50 50"
    width="40px"
    x="0px"
    y="0px"
  >
    <path
      d="M25.251,6.461c-10.318,0-18.683,8.365-18.683,18.683h4.068c0-8.071,6.543-14.615,14.615-14.615V6.461z"
      fill="#1E2A39"
    >
      <animatetransform
        attributeName="transform"
        attributeType="xml"
        dur="0.6s"
        from="0 25 25"
        repeatCount="indefinite"
        to="360 25 25"
        type="rotate"
      />
    </path>
  </svg>
</button>
`;

exports[`Button tests should render correctly when loaded 1`] = `
<button
  class="sc-bdVaJa gXMDQW"
  type="button"
>
  Button
</button>
`;
